pyproject.toml

```diff
@@ -93,7 +93,7 @@ optional = [
     'dask[array]>=2021.1.0,!=2024.8.0',
     'matplotlib>=3.7',
     'pooch>=1.6.0',
-    'pyamg',
+    'pyamg>=5.2',
```

Comment:
5.2 is SUPER new.

```
$ nep29 pyamg
| version |    date    |
| :-----: | :--------: |
|  5.2.1  | 2024-06-27 |
|  5.1.0  | 2024-03-19 |
|  5.0.0  | 2023-04-17 |
```

Reply:
I generally don't have aproblem with this, but I just wanted to point it out.

Reply:
I believe we need this version because of sparse dtype issues. As an optional, speed-enhancing dependency, I think it's OK.

Reply:
Ok